src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12906,5 +12951,20 @@ private new struct ResetPoint
             node = this.AddTrailingSkippedSyntax(node, trailingTrash.Node);
             return node;
         }
+
+        private SyntaxToken MergeTokens(SyntaxToken s1, SyntaxToken s2, SyntaxKind kind)
+        {
+            if (s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
+            {
+                s1 = SyntaxFactory.Token(s1.GetLeadingTrivia(), kind, s2.GetTrailingTrivia());
+            }
+            else
+            {
+                s2 = this.AddError(s2, ErrorCode.ERR_InvalidExprTerm, this.CurrentToken.Text);
+                AddTrailingSkippedSyntax(this.CreateMissingIdentifierName(), s2);
```

Comment:
>this.CreateMissingIdentifierName() [](start = 41, length = 34)

Why is this creating a missing identifier, and not just adding it to s1?

Reply:
Ah, I copied that off of another section of code. I'll remove that bit.